source/extensions/load_balancing_policies/override_host/load_balancer.cc

```diff
@@ -0,0 +1,363 @@
+                                        override_source.metadata_key.value());
+  }
+  // If neither host or metadata was found, return nullptr
+  return std::unique_ptr<SelectedHosts>(nullptr);
+}
+
+namespace {
+bool isIpv6Address(const SelectedHosts::Endpoint::Address& address) {
+  return absl::StrContains(address.address, ':');
+}
+
+std::string makeAddressKey(const SelectedHosts::Endpoint::Address& address) {
+  // IPv6 address needs to be wrapped in brackets.
+  if (isIpv6Address(address)) {
+    return absl::StrCat("[", address.address, "]:", address.port);
+  }
+  return absl::StrCat(address.address, ":", address.port);
+}
+} // namespace
+
+HostConstSharedPtr
+OverrideHostLoadBalancer::LoadBalancerImpl::findHost(const SelectedHosts::Endpoint& endpoint) {
+  HostMapConstSharedPtr hosts = priority_set_.crossPriorityHostMap();
+  if (hosts == nullptr) {
+    return nullptr;
+  }
+  std::string address_key = makeAddressKey(endpoint.address);
+
+  ENVOY_LOG(trace, "Looking up {} in {}", address_key,
+            absl::StrJoin(*hosts, ", ",
+                          [](std::string* out, Envoy::Upstream::HostMap::const_reference entry) {
+                            absl::StrAppend(out, entry.first);
+                          }));
+
+  if (const auto host_iterator = hosts->find(address_key); host_iterator != hosts->end()) {
+    // TODO(yanavlasov): Validate that host health status did not change.
+    return host_iterator->second;
+  }
+  return nullptr;
+}
+
+namespace {
+void updateFallbackIndexMetadata(::envoy::config::core::v3::Metadata& metadata,
+                                 uint32_t fallback_index) {
+  Envoy::ProtobufWkt::Struct fallback_index_metadata;
+  (*fallback_index_metadata.mutable_fields())[kEndpointsFallbackIndexFieldName].set_number_value(
+      fallback_index);
+  (*metadata.mutable_filter_metadata())[kEndpointsFallbackIndexKey] = fallback_index_metadata;
+}
+} // namespace
+
+HostConstSharedPtr OverrideHostLoadBalancer::LoadBalancerImpl::getEndpoint(
+    const SelectedHosts& selected_hosts, ::envoy::config::core::v3::Metadata& metadata) {
+  uint32_t fallback_index = 0;
+  if (!metadata.filter_metadata().contains(kEndpointsFallbackIndexKey)) {
+    // Use the primary endpoint.
+    ENVOY_LOG(trace, "Selecting primary endpoint {}", selected_hosts.primary.address.address);
+
+    // Endpoint extracted from the header does not have locality.
+    HostConstSharedPtr host = findHost(selected_hosts.primary);
+    // If the primary endpoint was found in the current host set, use it.
+    // Otherwise try to see if one of the failover endpoints is available. This
+    // is possible when the cluster received EDS update while the request to the
+    // endpoint picker was in flight.
+    if (host) {
+      // Save the first index into fallback hosts, so that subsequent calls to
+      // chooseHost method will use the fallback hosts.
+      updateFallbackIndexMetadata(metadata, 0);
+      return host;
+    }
```

Comment:
we needn't `updateFallbackIndexMetadata` here because the stream info will store the attemp count, please use that count directly to avoid unnecessary complexity.

Reply:
Done